src/array.jl

```diff
@@ -972,3 +972,31 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 writing/reading
+StructTypes.StructType(::Type{<:CategoricalVector}) = StructTypes.ArrayType()
```

Comment:
```suggestion
StructTypes.StructType(::Type{<:CategoricalArray}) = StructTypes.ArrayType()
```

Reply:
Does this make sense? One can't read or write multi-dimensional arrays to JSON.